syncer_test.go

```diff
@@ -29,6 +29,13 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// Cleanup helper to remove tmp files we've (maybe) created
+func cleanup(syncer *Syncer) {
+	for _, entry := range syncer.clients {
+		os.RemoveAll(entry.WriteDirectory)
```

Comment:
Are there warnings in the client configuration that this directory will be cleared after the tests, or a check to ensure that this directory doesn't already exist and have data before we run the tests?  Otherwise this could unexpectedly delete files that the test didn't create.